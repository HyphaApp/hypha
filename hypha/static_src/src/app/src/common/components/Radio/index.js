import * as React from "react";
import PropTypes from 'prop-types';
import HelperComponent from "@common/components/HelperComponent";


const DropDown = props => {
  return <div className="form__group ">
    <label htmlFor={props.name} className="form__question form__question--choice_field radio_select" required="">
      <span>{props.label}</span>
      {props.required ? <span className="form__required"> *</span> : ""}
    </label>
    
    <HelperComponent {...props.helperProps} />

    <div className="form__item">
      <ul className="grid grid--no-margin grid--two" >
        {(props.choices || []).map((choice, key) => {
          return <li className="form__item" key={key} onClick={e => props.onChange(props.name, choice[0])}>
          <input 
            type="radio" 
            name={props.name}
            value={choice[0]}
            checked={props.value === choice[0] ? "checked" : ""}
            onChange={e => props.onChange(props.name, choice[0])}
          />
          <label className="form__label" htmlFor={`${props.name}-${key}`}>{choice[1]}</label>
        </li>;
        })}
      </ul>

    </div>
  </div>
}

DropDown.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  required: PropTypes.bool,
  onChange: PropTypes.func,
  value: PropTypes.node,
  helperProps: PropTypes.object,
  choices: PropTypes.array,
}

DropDown.displayName = 'DropDown';
export default DropDown;
