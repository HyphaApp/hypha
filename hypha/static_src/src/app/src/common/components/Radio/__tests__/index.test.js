import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import Radio from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test radio component", () => {
  const name = "test name";
  const label = "test label";
  const required = false;
  const choices = [["1", "one"], ["2", "two"]];
  const helperProps = {};
  const value = "1";
  const onChange = jest.fn();

  const subject = mount(<Radio
    name={name}
    label={label}
    required={required}
    choices={choices}
    value={value}
    helperProps={helperProps}
    onChange={onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and span element with passed text", () => {
    expect(subject.find('.form__group').length).toBe(1);
    expect(subject.children().length).toEqual(1);
    expect(subject.find('span').text()).toBe(label);
    expect(subject.find('.form__required').length).toBe(0);
    subject.find('li').first().props().onClick();
    expect(onChange).toHaveBeenCalledWith(name, "1")
    expect(onChange).toHaveBeenCalled();
    subject.find('input').last().props().onChange();
    expect(onChange).toHaveBeenCalledWith(name, "2")
    expect(onChange).toHaveBeenCalledTimes(2);
  });

  test('render a radio component', () => {
    expect(subject).toMatchSnapshot();
  });

});

describe("Test radio component with required & empty choices", () => {
  const name = "test name";
  const label = "test label";
  const required = true;
  const choices = [];
  const helperProps = {};
  const value = "1";
  const onChange = jest.fn();

  const subject = mount(<Radio
    name={name}
    label={label}
    required={required}
    choices={choices}
    value={value}
    helperProps={helperProps}
    onChange={onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and span element with passed text", () => {
    expect(subject.find('.form__group').length).toBe(1);
    expect(subject.children().length).toEqual(1);
    expect(subject.find('span').first().text()).toBe(label);
    expect(subject.find('.form__required').length).toBe(1);
    expect(subject.find('li').length).toBe(0);
    expect(subject.find('input').length).toBe(0);
    expect(choices.length).toBe(0);
  });

  test('render a radio component with required & empty choices', () => {
    expect(subject).toMatchSnapshot();
  });

});
