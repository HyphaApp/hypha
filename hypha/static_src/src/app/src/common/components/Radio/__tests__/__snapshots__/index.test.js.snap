// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test radio component render a radio component 1`] = `
<Radio
  choices={
    Array [
      Array [
        "1",
        "one",
      ],
      Array [
        "2",
        "two",
      ],
    ]
  }
  helperProps={Object {}}
  label="test label"
  name="test name"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "test name",
          "1",
        ],
        Array [
          "test name",
          "2",
        ],
      ],
    }
  }
  required={false}
  value="1"
>
  <div
    className="form__group "
  >
    <label
      className="form__question form__question--choice_field radio_select"
      htmlFor="test name"
      required=""
    >
      <span>
        test label
      </span>
    </label>
    <HelperComponent />
    <div
      className="form__item"
    >
      <ul
        className="grid grid--no-margin grid--two"
      >
        <li
          className="form__item"
          key="0"
          onClick={[Function]}
        >
          <input
            checked="checked"
            name="test name"
            onChange={[Function]}
            type="radio"
            value="1"
          />
          <label
            className="form__label"
            htmlFor="test name-0"
          >
            one
          </label>
        </li>
        <li
          className="form__item"
          key="1"
          onClick={[Function]}
        >
          <input
            checked=""
            name="test name"
            onChange={[Function]}
            type="radio"
            value="2"
          />
          <label
            className="form__label"
            htmlFor="test name-1"
          >
            two
          </label>
        </li>
      </ul>
    </div>
  </div>
</Radio>
`;

exports[`Test radio component with required & empty choices render a radio component with required & empty choices 1`] = `
<Radio
  choices={Array []}
  helperProps={Object {}}
  label="test label"
  name="test name"
  onChange={[MockFunction]}
  required={true}
  value="1"
>
  <div
    className="form__group "
  >
    <label
      className="form__question form__question--choice_field radio_select"
      htmlFor="test name"
      required=""
    >
      <span>
        test label
      </span>
      <span
        className="form__required"
      >
         *
      </span>
    </label>
    <HelperComponent />
    <div
      className="form__item"
    >
      <ul
        className="grid grid--no-margin grid--two"
      />
    </div>
  </div>
</Radio>
`;
